src/Streamly/Internal/Data/Stream/StreamD/Eliminate.hs

```diff
@@ -193,6 +194,70 @@ parselMx' pstep initial extract (Stream step state) = do
             PR.Done _ b -> return b
             PR.Error err -> throwM $ ParseError err
 
+-- | Run a 'Parse' over a stream and return rest of the Stream.
+{-# INLINE_NORMAL parseAndReturnD #-}
+parseAndReturnD
```

Comment:
If this is as efficient as `parse`, you should implement `parse` interms of this.

Reply:
You can probably omit the suffix `D`. This is the `StreamD` module.